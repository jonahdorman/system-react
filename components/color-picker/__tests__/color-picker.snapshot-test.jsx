import React from 'react';
import renderer from 'react-test-renderer';

import ColorPicker from '../../color-picker';

test('ColorPicker Base DOM Snapshot', () => {
	const domTree = renderer
		.create(<ColorPicker className="test_class_name" />)
		.toJSON();
	expect(domTree).toMatchSnapshot();
});

test('ColorPicker Menu Open DOM Snapshot', () => {
	const domTree = renderer
		.create(<ColorPicker classNameMenu="test_class_name_menu" isOpen />)
		.toJSON();
	expect(domTree).toMatchSnapshot();
});
