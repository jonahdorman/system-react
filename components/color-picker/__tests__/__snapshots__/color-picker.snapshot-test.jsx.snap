// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorPicker Base DOM Snapshot 1`] = `
<div
  className="slds-color-picker test_class_name"
>
  <div
    className="slds-color-picker__summary"
  >
    <span
      className="slds-color-picker__summary-label"
      htmlFor="color-picker-summary-input-r1iIHtt97"
    />
    <button
      className="slds-button slds-button_icon-more slds-color-picker__summary-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="slds-swatch"
        style={
          Object {
            "backgroundColor": "",
            "backgroundImage": "linear-gradient(45deg, white 40%, red 0, red 60%, white 0)",
            "border": "1px solid #cccccc",
          }
        }
      >
        <span
          className="slds-assistive-text"
        >
          
        </span>
      </span>
      <svg
        aria-hidden="true"
        className="slds-button__icon slds-button__icon_x-small slds-m-left_xx-small"
      >
        <use />
      </svg>
    </button>
    <div
      className="slds-form-element slds-color-picker__summary-input"
    >
      <div
        className="slds-form-element__control"
      >
        <input
          aria-describedby="color-picker-summary-error-r1iIHtt97"
          className="slds-input"
          id="color-picker-summary-input-r1iIHtt97"
          maxLength="7"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    
  </div>
</div>
`;

exports[`ColorPicker Menu Open DOM Snapshot 1`] = `
<div
  className="slds-color-picker"
>
  <div
    className="slds-color-picker__summary"
  >
    <span
      className="slds-color-picker__summary-label"
      htmlFor="color-picker-summary-input-SybiUBKt5m"
    />
    <button
      className="slds-button slds-button_icon-more slds-color-picker__summary-button"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      <span
        className="slds-swatch"
        style={
          Object {
            "backgroundColor": "",
            "backgroundImage": "linear-gradient(45deg, white 40%, red 0, red 60%, white 0)",
            "border": "1px solid #cccccc",
          }
        }
      >
        <span
          className="slds-assistive-text"
        >
          
        </span>
      </span>
      <svg
        aria-hidden="true"
        className="slds-button__icon slds-button__icon_x-small slds-m-left_xx-small"
      >
        <use />
      </svg>
    </button>
    <div
      className="slds-form-element slds-color-picker__summary-input"
    >
      <div
        className="slds-form-element__control"
      >
        <input
          aria-describedby="color-picker-summary-error-SybiUBKt5m"
          className="slds-input"
          id="color-picker-summary-input-SybiUBKt5m"
          maxLength="7"
          onChange={[Function]}
          type="text"
          value=""
        />
      </div>
    </div>
    <div
      className="absolute-positioned slds-color-picker__selector slds-popover test_class_name_menu"
      onKeyDown={[Function]}
      style={
        Object {
          "outline": 0,
          "pointerEvents": "none",
          "position": "absolute",
        }
      }
    >
      <div
        className="slds-popover__body"
      >
        <div
          className="slds-tabs_default"
          data-tabs={true}
          id="B1QoLBFK57"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <ul
            className="slds-tabs_default__nav"
            id="B1QoLBFK57-slds-tabs__nav"
            role="tablist"
          >
            <li
              aria-controls="B1QoLBFK57-slds-tabs_panel-0"
              aria-selected="true"
              className="slds-text-title_caps slds-active slds-tabs_default__item"
              id="B1QoLBFK57-slds-tabs_tab-0"
              role="tab"
              tabIndex="0"
              title="Default"
            >
              <a
                className="slds-active slds-tabs_default__link"
                href="javascript:void(0);"
                role="presentation"
                tabIndex="-1"
              >
                Default
              </a>
            </li>
            <li
              aria-controls="B1QoLBFK57-slds-tabs_panel-1"
              aria-selected="false"
              className="slds-text-title_caps slds-tabs_default__item"
              id="B1QoLBFK57-slds-tabs_tab-1"
              role="tab"
              tabIndex={null}
              title="Custom"
            >
              <a
                className="slds-tabs_default__link"
                href="javascript:void(0);"
                role="presentation"
                tabIndex="-1"
              >
                Custom
              </a>
            </li>
          </ul>
          <div
            aria-labelledby="B1QoLBFK57-slds-tabs_tab-0"
            className="slds-show slds-tabs_default__content"
            id="B1QoLBFK57-slds-tabs_panel-0"
            role="tabpanel"
          >
            <ul
              className="slds-color-picker__swatches"
              role="listbox"
            >
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#e3abec",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #e3abec
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#c2dbf7",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #c2dbf7
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#9fd6ff",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #9fd6ff
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#9de7da",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #9de7da
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#9df0c0",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #9df0c0
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#fff099",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #fff099
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#fed49a",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #fed49a
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#d073e0",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #d073e0
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#86baf3",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #86baf3
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#5ebbff",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #5ebbff
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#44d8be",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #44d8be
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#3be282",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #3be282
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#ffe654",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #ffe654
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#ffb758",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #ffb758
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#bd35bd",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #bd35bd
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#5779c1",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #5779c1
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#5679c0",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #5679c0
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#00aea9",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #00aea9
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#3cba4c",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #3cba4c
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#f5bc25",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #f5bc25
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#f99221",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #f99221
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#580d8c",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #580d8c
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#001970",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #001970
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#0a2399",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #0a2399
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#0b7477",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #0b7477
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#0b6b50",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #0b6b50
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#b67e11",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #b67e11
                    </span>
                  </span>
                </a>
              </li>
              <li
                className="slds-color-picker__swatch"
                role="presentation"
              >
                <a
                  className="slds-color-picker__swatch-trigger"
                  onClick={[Function]}
                  role="option"
                  tabIndex={0}
                >
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "#b85d0d",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      #b85d0d
                    </span>
                  </span>
                </a>
              </li>
            </ul>
          </div>
          <div
            aria-labelledby="B1QoLBFK57-slds-tabs_tab-1"
            className="slds-hide slds-tabs_default__content"
            id="B1QoLBFK57-slds-tabs_panel-1"
            role="tabpanel"
          >
            <div
              className="slds-color-picker__custom"
            >
              <div>
                <p
                  className="slds-assistive-text"
                  id="color-picker-instructions-SybiUBKt5m"
                >
                  Use arrow keys to select a saturation and brightness, on an x and y axis.
                </p>
                <div
                  className="slds-color-picker__custom-range"
                  onClick={[Function]}
                  role="presentation"
                  style={
                    Object {
                      "background": "hsl(0, 100%, 50%)",
                    }
                  }
                >
                  <a
                    aria-atomic="true"
                    aria-describedby="color-picker-instructions-SybiUBKt5m"
                    aria-live="assertive"
                    className="slds-color-picker__range-indicator"
                    onKeyDown={[Function]}
                    role="button"
                    style={
                      Object {
                        "bottom": "0%",
                        "left": "0%",
                      }
                    }
                    tabIndex={0}
                  />
                </div>
                <div
                  className="slds-color-picker__hue-and-preview"
                >
                  <label
                    className="slds-assistive-text"
                    htmlFor="color-picker-input-range-SybiUBKt5m"
                  >
                    Select Hue
                  </label>
                  <input
                    className="slds-color-picker__hue-slider"
                    id="color-picker-input-range-SybiUBKt5m"
                    max="360"
                    min="0"
                    onChange={[Function]}
                    type="range"
                    value={0}
                  />
                  <span
                    className="slds-swatch"
                    style={
                      Object {
                        "backgroundColor": "",
                        "backgroundImage": "linear-gradient(45deg, white 40%, red 0, red 60%, white 0)",
                        "border": "1px solid #cccccc",
                      }
                    }
                  >
                    <span
                      className="slds-assistive-text"
                    >
                      
                    </span>
                  </span>
                </div>
              </div>
              <div
                className="slds-color-picker__custom-inputs"
              >
                <div
                  className="slds-form-element slds-color-picker__input-custom-hex"
                >
                  <label
                    className="slds-form-element__label"
                    htmlFor="color-picker-input-hex-SybiUBKt5m"
                  >
                    Hex
                  </label>
                  <div
                    className="slds-form-element__control"
                  >
                    <input
                      aria-describedby="color-picker-custom-error-SybiUBKt5m"
                      className="slds-input"
                      id="color-picker-input-hex-SybiUBKt5m"
                      maxLength="7"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                </div>
                <div
                  className="slds-form-element slds-color-picker__input-custom-r"
                >
                  <label
                    className="slds-form-element__label"
                    htmlFor="color-picker-input-r-SybiUBKt5m"
                  >
                    R
                  </label>
                  <div
                    className="slds-form-element__control"
                  >
                    <input
                      aria-describedby="color-picker-custom-error-SybiUBKt5m"
                      className="slds-input"
                      id="color-picker-input-r-SybiUBKt5m"
                      maxLength="3"
                      onChange={[Function]}
                      type="text"
                      value="0"
                    />
                  </div>
                </div>
                <div
                  className="slds-form-element slds-color-picker__input-custom-g"
                >
                  <label
                    className="slds-form-element__label"
                    htmlFor="color-picker-input-g-SybiUBKt5m"
                  >
                    G
                  </label>
                  <div
                    className="slds-form-element__control"
                  >
                    <input
                      aria-describedby="color-picker-custom-error-SybiUBKt5m"
                      className="slds-input"
                      id="color-picker-input-g-SybiUBKt5m"
                      maxLength="3"
                      onChange={[Function]}
                      type="text"
                      value="0"
                    />
                  </div>
                </div>
                <div
                  className="slds-form-element slds-color-picker__input-custom-b"
                >
                  <label
                    className="slds-form-element__label"
                    htmlFor="color-picker-input-b-SybiUBKt5m"
                  >
                    B
                  </label>
                  <div
                    className="slds-form-element__control"
                  >
                    <input
                      aria-describedby="color-picker-custom-error-SybiUBKt5m"
                      className="slds-input"
                      id="color-picker-input-b-SybiUBKt5m"
                      maxLength="3"
                      onChange={[Function]}
                      type="text"
                      value="0"
                    />
                  </div>
                </div>
              </div>
              
            </div>
          </div>
        </div>
      </div>
      <footer
        className="slds-popover__footer"
      >
        <div
          className="slds-color-picker__selector-footer"
        >
          <button
            className="slds-button slds-button_neutral slds-color-picker__selector-cancel"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Cancel
          </button>
          <button
            className="slds-button slds-button_brand slds-color-picker__selector-submit"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Done
          </button>
        </div>
      </footer>
    </div>
    
  </div>
</div>
`;
